saleor/payment/gateways/np_atobarai/api.py

```diff
@@ -0,0 +1,209 @@
+            return (
+                f"{ad.country_area}"
+                f"{jap_ad.city}"
+                f"{jap_ad.neighborhood}"
+                f"{ad.street_address_2}"
+                f"{ad.street_address_1}"
+            )
+
+
+def register_transaction(
+    config: ApiConfig, payment_information: "PaymentData"
+) -> PaymentResult:
+    order_date = timezone.now().strftime("%Y-%m-%d")
+
+    billing = payment_information.billing
+    shipping = payment_information.shipping
+
+    if not billing:
+        raise PaymentError(
+            "Billing address is required for transaction in NP Atobarai."
+        )
+    if not shipping:
+        raise PaymentError(
+            "Shipping address is required for transaction in NP Atobarai."
+        )
+
+    data = {
+        "transactions": [
+            {
+                "shop_transaction_id": payment_information.payment_id,
+                "shop_order_date": order_date,
+                "settlement_type": NP_ATOBARAI,
+                "billed_amount": int(
+                    price_to_minor_unit(
+                        payment_information.amount, payment_information.currency
+                    )
+                ),
+                "customer": {
+                    "customer_name": billing.first_name,
+                    "company_name": billing.company_name,
+                    "zip_code": billing.postal_code,
+                    "address": _format_address(billing),
+                    "tel": billing.phone.replace("+81", "0"),
+                    "email": payment_information.customer_email,
+                },
+                "dest_customer": {
+                    "customer_name": _format_name(shipping),
+                    "company_name": shipping.company_name,
+                    "zip_code": shipping.postal_code,
+                    "address": _format_address(shipping),
+                    "tel": shipping.phone.replace("+81", "0"),
+                },
+                "goods": [
+                    {
+                        "quantity": line.quantity,
+                        "goods_name": line.description,
+                        "goods_price": int(
+                            price_to_minor_unit(
+                                line.gross, payment_information.currency
+                            )
+                        ),
+                    }
+                    for line in payment_information.lines
+                ],
+            },
+        ]
+    }
+
+    response = np_request(config, "post", "/transactions", json=data)
+    response_data = response.json()
+
+    if "results" in response_data:
+        transaction = response_data["results"][0]
+        status = transaction["authori_result"]
+        transaction_id = transaction["np_transaction_id"]
+        errors = []
+
+        if status == PaymentStatus.PENDING:
+            cancel_error_codes = _cancel(config, transaction_id)
+            if cancel_error_codes:
+                logger.error(
```

Comment:
Is it something that should be logged as an error? This will be sent to the sentry. Not sure if this is something that we should log here. Maybe warning would be better 🤔  as this is an expected case that some transactions can be canceled.

Reply:
This error is logged only if NP doesn't cancel the payment that is in pending state. Pending payment needs to be resolved either by cancelling it, or by contacting NP customer support. Leaving that payment in pending state indefinitely is an error, so I think it's good idea to log it as error.